Source/WebCore/dom/Document.h

```diff
@@ -2583,6 +2587,8 @@ class Document
 
     const std::optional<FrameIdentifier> m_frameIdentifier;
     std::optional<bool> m_cachedCookiesEnabled;
+
+    mutable std::unique_ptr<CSSParserContext> m_cachedCSSParserContext;
```

Comment:
If we are caching at the document level, can we get rid of the caching in `HTMLCanvasElement`?

Reply:
Maybe? Probably? I don't know enough about canvas or why that cache was added.

Reply:
@heycam HTMLCanvasElement::cssParserContext() question